core/src/main/scala/cats/data/Newtype.scala

```diff
@@ -0,0 +1,8 @@
+package cats
+package data
+
+trait Newtype { self =>
```

Comment:
Can you add some documentation that motivates this `trait`?

Reply:
Yes, good idea :)